bulk-upload-resources-python-example/upload_resources.py

```diff
@@ -0,0 +1,85 @@
+import sys
+import pandas
+import numpy as np
+from configparser import RawConfigParser
+from datetime import datetime
+import json
+import requests
+
+supported_timestamps = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%M:%S.%fZ']
+
+if len(sys.argv) < 2:
+    print("Usage: `python ./main.py <csv-path> <configuration-path>`")
+    exit()
+
+csv_file_path = sys.argv[1]
+config_path = sys.argv[2]
+
+config = RawConfigParser()
+config.read(config_path)
+
+
+def get_auth(config):
+    """ Get the authentication out of our config file """
+    return config.get("domain", "client-key"), config.get("domain", "client-secret")
+
+
+def convert_datetime_to_epoch(datetime_string):
+    """ Converts the datetime to epoch in miliseconds there are two possible time formats"""
+    try:
+        return datetime.strptime(datetime_string, '%Y-%m-%dT%H:%M:%SZ').timestamp() * 1000
+    except:
+        return datetime.strptime(datetime_string, '%Y-%m-%dT%H:%M:%S.%fZ').timestamp() * 1000
+
+
+def transform_data(data):
+    """Transforms data from the csv format to the expected format bij the api"""
+    data['timestamp'] = data['timestamp'].apply(
+        lambda x: convert_datetime_to_epoch(x))
+    data['metric'] = data['metric'].apply(lambda metric: metric.replace('waylay.resourcemessage.metric.', ''))
+    return data
+
+
+def chunck_to_message(chunck):
+    """Makes entry in the dict with the key the value of the metric and with as value the value from de value key.
+    Also drops the unneeded entries"""
+
+    for row in chunck:
+        row[row['metric']] = row.pop('value')
+        row.pop('metric')
+    return chunck
+
+
+def send_messages(data_json_array, config: RawConfigParser):
+    """Sends the json array to the waylay vendor in chunks of specific length (as defined in config)"""
+    auth = get_auth(config)
+    chunck_size = int(config.get('domain', 'values-per-post'))
```

Comment:
Good idea to add this to the config! But maybe this should not be under `domain`, but under `network` or something like that.